.github/workflows/documentation.yml

```diff
@@ -0,0 +1,33 @@
+name: Build documentation
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - 'docs/**'
+      - 'mkdocs.yml'
+  pull_request:
+    paths:
+      - 'docs/**'
+      - 'mkdocs.yml'
+
+jobs:
+  build:
+    name: Build documentation
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: mkdocs build --strict
+      run: |
+        docker run --rm -v ${PWD}:/docs squidfunk/mkdocs-material build --strict --config-file docs/mkdocs.yml
+  deploy:
+    name: Deploy documentation to gh-pages
+    runs-on: ubuntu-20.04
+    if: github.event_name != 'pull_request'
+    needs: build
+    steps:
+    - uses: actions/checkout@v2
+    - name: mkdocs gh-pages --force
```

Comment:
Are you sure you want to deploy this way still? Wouldn't the more generic action I used be useful? This might not work since the context changed, you'll need to pass ENV through to docker that it expects, at which point it is probably better to just do a build in a separate step and have the action I mention deploy it, where deploy config is more explicit.

Reply:
Good point. 
Keep in mind that i just started and anything is subject to change. Will update the description soon and then we can work on the tasklist

Reply:
Tested it out, but run into issues as i need to chache the build between jobs. If using this action requires caching i would vote for using the mkdocs gh-deploy as it is now